### PR TITLE
Debug console - SAML support

### PR DESCRIPTION
**Pull request by sdesen**
Screenshot 2024-05-20 at 12 32 38 PM
Screenshot 2024-05-20 at 12 34 27 PM
Screenshot 2024-05-20 at 12 34 41 PM

### CODE REVIEW

File: authorization-server/jwt-authorization-grant-token-exchange.js [hunk 1]
```diff
@@ -29,12 +29,11 @@ export async function authorizationGrantTokenExchange(ctx, configuration, redisC
   }
 
   // Lookup by payload.sub and get the last id assertion saved
-  const { subjectToken } = getSubjectToken(payload.sub, provider.client_id);
-
-  // TODO store in redis for debug console
+  const { subjectToken, tokenType } = getSubjectToken(payload.sub, provider.client_id);
 
   await redisClient.set(`jag_subject_token:${payload.sub}`, subjectToken);
```
Comment (neelav) [inline authorization-server/jwt-authorization-grant-token-exchange.js:34]: i forgot to mention in the previous review.  Can you add a comment saying we store this info to redis purely for informational purposes to surface to the debug console?

File: wiki0/src/client/components/home.tsx [hunk 2]
```diff
@@ -108,11 +115,21 @@ function Home() {
         </div>
 
         <div className="space-y-4 ">
+          {/* <DebugCard>
```
Comment (neelav) [inline wiki0/src/client/components/home.tsx:118]: delete?

File: wiki0/src/client/components/tokenViewer.tsx [hunk 3]
```diff
@@ -4,13 +4,44 @@ import { parseJwt } from '../utils';
...
+const defaultProps = {
+  saml: false,
+};
+
+function TokenViewer({ token, saml }: TokenViewerProps) {
   if (!token) {
     return <div className="text-gray-400">No JWT token found</div>;
   }
 
+  const displayToken = () => {
```
Comment (neelav) [inline wiki0/src/client/components/tokenViewer.tsx:19]: nit: pass in the token variable and move this function to the top level

File: wiki0/src/server/controllers/authtoken.ts [hunk 4]
```diff
@@ -17,10 +15,21 @@ controller.get('/', async (req, res) => {
   });
 
   // Tack on hardcoded request body for debug console demonstration purposes
+  const savedSubjectToken = await req.app.locals.redisClient.get(
```
Comment (neelav) [inline wiki0/src/server/controllers/authtoken.ts:18]: also add to the comment above that these values were set by the authorization server

File: wiki0/src/server/main.ts [hunk 5]
```diff
@@ -23,11 +23,13 @@ app.use(express.json());
...
 
-const redisClient = createClient({
-  url: process.env.REDIS_SERVER,
-});
+const redisClient = createClient();
+
+redisClient.on('error', (error) => console.error(`Redis Error : ${error}`));
 
-redisClient.connect().catch(console.error);
+(async () => {
```
[the remark this answers is not in the document]
Comment (neelav) [reply wiki0/src/server/main.ts:30]: hmm what's the error?  The old and new code are essentially equivalent.

File: wiki0/src/server/main.ts [hunk 6]
```diff
@@ -23,11 +23,13 @@ app.use(express.json());
 // TODO uncomment when ready
 // app.use('/.well-known', wellknown);
 
-const redisClient = createClient({
-  url: process.env.REDIS_SERVER,
-});
+const redisClient = createClient();
```
Comment (neelav) [inline wiki0/src/server/main.ts:26]: what happened to the passed in url?

File: wiki0/src/client/components/editpage.tsx [hunk 7]
```diff
@@ -15,6 +15,14 @@ type RequestLogItem = {
 
 const API_BASE_URL = '/api/articles';
 
+const getParsedTokenString = (headers: string) => {
+  const parts = headers.split('Bearer ');
+  if (parts?.length === 2) {
+    return parts[1]?.replaceAll('"', '')?.replaceAll('}', '');
```
Comment (kamronbatman) [inline wiki0/src/client/components/editpage.tsx:21]: Whats the token here? It's not base64?

File: wiki0/src/client/components/tokenViewer.tsx [hunk 8]
```diff
@@ -4,13 +4,44 @@ import { parseJwt } from '../utils';
...
+          <pre className="">{atob(token)}</pre>
+        </div>
+      );
+    }
+    try {
+      tokenJson = JSON.stringify(parseJwt(token), null, 2);
+    } catch (e) {
+      tokenJson = 'Error parsing token';
+    }
+    return tokenJson;
```
Comment (kamronbatman) [inline wiki0/src/client/components/tokenViewer.tsx:42]: nit, can be cleaned up
```suggestion
    if (saml) {
      return (
        <div className="max-w-md">
          Tip: Copy and Paste into{' '}
          <a
            target="_blank"
            href="https://samltool.io"
            className="font-medium text-blue-600 dark:text-blue-500 hover:underline"
            rel="noreferrer"
          >
            samltool.io
          </a>
          <pre className="">{atob(token)}</pre>
        </div>
      );
    }
    try {
      return JSON.stringify(parseJwt(token), null, 2);
    } catch (e) {
      return 'Error parsing token';
    }
```

File: wiki0/src/server/main.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (kamronbatman) [reply wiki0/src/server/main.ts:30]: In this context where we fire/forget in an async function, I am not sure awaiting it inside the function is any better than not awaiting it at the top-level execution. 🤔

File: wiki0/src/client/components/editpage.tsx (hunk 7 above)
Comment (kamronbatman) [reply wiki0/src/client/components/editpage.tsx:21]: Oh that is super weird that it has a `}"` at the end. (Bug?)

Comment (kamronbatman) [reply wiki0/src/client/components/editpage.tsx:21]: Maybe add a comment? `// TODO: Token building is probably buggy, will look into and clean up later`

File: authorization-server/jwt-authorization-grant-token-exchange.js (hunk 1 above)
Comment (neelav) [reply authorization-server/jwt-authorization-grant-token-exchange.js:34]: i was referring to a new comment like the one I mentioned in my github comment

File: wiki0/src/server/controllers/authtoken.ts (hunk 4 above)
Comment (neelav) [reply wiki0/src/server/controllers/authtoken.ts:18]: please update comment

**Review by kamronbatman — APPROVED** [review]

**Pull request merged**